Bandwidthd 18.06: fix undefined references to inline functions

padre-lacroix opened the pull request:
Maintainer: Jean-Michel Lacroix / @padre-lacroix
Compile tested: i386_pentium4 OpenWrt 18.06
Run tested: OpenWrt 18.06.1 r7258-5eb055306f, QEMU Virtual CPU version (cpu64-rhel6).  Able to generate the graphs as expected.

Signed-off-by: Jean-Michel Lacroix <lacroix@lepine-lacroix.info>

Description:
This is the same commit that took place in master 3 weeks ago.
gcc-7 with -Os makes inline functions disappeard. It is caused by
the new C11 inline semantics. pass option -fgnu89-inline to gcc let
it use gnu inline semantics.
see https://wiki.debian.org/GCC7#Porting_help

This will allow to have bandwidthd in 18.06.

padre-lacroix commented:
I will check.  I also have to correct the title in the commit as it seems that it is not OK as it does not start with the package name (I have not submitted anything for a while and the rules changed it seems).  I will not be able to check for a couple of days.

padre-lacroix commented:
I tested a few things:
EXTRA_CFLAGS+= $(TARGET_CPPFLAGS) -fgnu89-inline
or 
EXTRA_CFLAGS+= $(TARGET_CPPFLAGS) -std=gnu89 do not make any difference in the compilation.  I tested with make -j1 V=s and got the same output.  I run tested the one with EXTRA_CFLAGS+= $(TARGET_CPPFLAGS) -fgnu89-inline and its works fine.  I have not tested the one with EXTRA_CFLAGS+= $(TARGET_CPPFLAGS) -std=gnu89, but I will later tomorrow.
I also tested:
EXTRA_CFLAGS+= $(TARGET_CPPFLAGS)
EXTRA_CFLAGS+= $(TARGET_CFLAGS) -std=gnu89
and this compiles also without a problem.  I have not run tested that one either: I will test it tomorrow and if it works I will update the pull request with this, unless someone has a reason not to.

padre-lacroix commented:
OK,  So I made the following change: I added the following line:
TARGET_CFLAGS += -std=gnu89.

I also corrected the title in the commit to have the package name first.

I tested my compiled bin and it woks fine.  All the various options tested that compiled gave a file with the same exact size.

I hope that it is OK now.  If these is anything else, please let me know.

I hope that it is OK now.

padre-lacroix commented:
I see that the Travis CI build failed.  But, by looking at the Details, I do not see why it failed, just it exited with a status of 1.

Any suggestions?

uxgood commented:
it breaks after postgresql host-compile.I think postgresql is broken.

pprindeville commented:
> What are my options here?

You could wait until postgresql gets fixed and rebuild then.

padre-lacroix commented:
The thing is that the package postgresql in openwrt-18.06 is does not seem damaged: it compiles fine on my system and is available as a download on a OpenWrt 18.06.1 router.  I can compile all applications for i386_pentium4:
libpq, pgsql-cli, pgsql-cli-extra and pgsql-server.

I wrote, last Friday, to the maintainer of the prosgresql package to inquire if the package was damaged, but I did not get an answer yet.

neheb commented:
It's probably Travis acting up. I would ignore and merge.

padre-lacroix commented:
"Can you rebase on latest commit, it should trigger CircleCi"
Not sure what to do here.  I did git rebase -i HEAD~1 (answer: Successfully rebased and updated refs/heads/bandwidthd-18.06.) and a git push -f (answer: Everything up-to-date).  But nothing is hapenning!
Sorry here for my ignorance!

neheb commented:
git checkout master
git pull
git checkout branch
git rebase master

champtar commented:
@neheb this PR is for 18.06

champtar commented:
@padre-lacroix try @neheb instructions with 'openwrt-18.06' instead of 'master'

padre-lacroix commented:
Did:
[lacroix@centos bandwidthd]$ git checkout openwrt-18.06
Switched to branch 'openwrt-18.06'
[lacroix@centos bandwidthd]$ git pull
Already up-to-date.
[lacroix@centos bandwidthd]$ git checkout bandwidthd-18.06
Switched to branch 'bandwidthd-18.06'
[lacroix@centos bandwidthd]$ git rebase openwrt-18.06
Current branch bandwidthd-18.06 is up to date.
[lacroix@centos bandwidthd]$ git push -f
Everything up-to-date

padre-lacroix commented:
Maybe it would just be easier to close the pull request and start over!

neheb commented:
Note that the master and openwrt-18.06 branches on your repo at https://github.com/padre-lacroix/packages/tree/openwrt-18.06 are not in sync with the main one. I usually do

git clone https://github.com/openwrt/packages.git
cd packages
git remote add neheb https://github.com/neheb/packages
git checkout -b test
git push --force neheb

This way when I checkout master it's master to the main repository and not my fork.

champtar commented:
@padre-lacroix it might be easier, but you have 2 guys ready to teach you ;)
can you paste the output of
git remote -v

champtar commented:
@pprindeville can you confirm the patch is ok for you (and validate your review)

padre-lacroix commented:
lacroix@centos bandwidthd]$ git remote -v
origin	https://github.com/padre-lacroix/packages (fetch)
origin	https://github.com/padre-lacroix/packages (push)
upstream	https://github.com/openwrt/packages (fetch)
upstream	https://github.com/openwrt/packages (push)

padre-lacroix commented:
OK:
The instructions above from @champtar for the rebase worked: thank you very much for your help, and also @neheb .
The good news is that the Travis CI built is now working.
The bad news if that the buildpr is failing, and there is no information as to why.

champtar commented:
@padre-lacroix you have to follow a long trail of links but there is all the details for CircleCI
```
env: 'time': No such file or directory
```
I'll fix the CircleCI image later today

pprindeville commented:
Assuming that the remote is called “upstream”:

% git fetch -a upstream
% git pull —rebase upstream openwrt-18.06
% git push -f -u origin mybranch

Or:

% git checkout master
% git fetch -a upstream
% git reset —hard upstream/master
% git push -u origin master
% git checkout mybranch
% git pull —rebase origin master
% git push -f -u origin my branch

More typically when working off master.

champtar commented:
@padre-lacroix i've updated the CircleCI image, can you rerebase please

padre-lacroix commented:
Well: it seems to have worked now!
Thanks everybody @champtar , @neheb , @pprindeville 
Jean-Michel Lacroix

champtar commented:
Thanks @padre-lacroix
